### ISSUE TITLE
Implement Sql Catalog.

### ISSUE THREAD

**Issue by liurenjie1024** [apache/iceberg-rust#248]
Implement sql catalog.

**Comment by hendrikmakait** [comment]
Is there anything missing or has this been completed by #524?

**Comment by kevinjqliu** [comment]
@hendrikmakait i dont think so, i see a lot of `todos` in that PR https://github.com/apache/iceberg-rust/pull/524/files#diff-0095ccc22a02ff26cc0c2f2016a0573cecfeded94dfeb7fdf249b2ec57764194R186
